eventsourcing/src/main/java/org/axonframework/eventsourcing/eventstore/GlobalIndexPosition.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing.eventstore;
+
+import jakarta.annotation.Nonnull;
+
+/**
+ * An immutable implementation of {@link Position} which represents positions
+ * with an index in the global event stream.
+ *
+ * @author John Hendrikx
+ * @since 5.0.0
+ */
+public final class GlobalIndexPosition implements Position {
+
+    private static final long MINIMUM_INDEX = Long.MIN_VALUE;
```

Comment:
Why you have changed MINIMUM_INDEX form -1 to Long.MIN_VALUE? I don't expect at all events stored with minus, so the position of -1 is a Position BEFORE the first event. If I understand it correctly.